src/devices/machine/ncr53c90.cpp

```diff
@@ -1329,3 +1359,36 @@ void ncr53c94_device::check_drq()
 	else
 		ncr53c90_device::check_drq();
 }
+
+void ncr53c94_device::decrement_tcounter(int count)
+{
+	if (!dma_command)
+		return;
+
+	const uint32_t tcounter_mask = (config2 & S2FE) ? 0xffffff : 0xffff;
+	if (!(status & S_TC0))
+		tcounter = (tcounter - count) & tcounter_mask;
+	else
+		tcounter = 0;
+
+	if (tcounter == 0)
+		status |= S_TC0;
+
+	check_drq();
+}
```

Comment:
As far as I can tell, this is identical to the other version with the exception of the mask values and the absence of the comment in this one.  Would it be better to make the mask a data member so only one version of this function is necessary, and make `conf2_w` virtual so the derived class can update the mask when necessary?

Reply:
Yeah, that's pretty much it. That sounds like a sensible way to remove the code duplication. I pushed a new commit to use a common `tcounter_mask` that gets updated by `conf2_w` as suggested.